pkg/operator/api/userconfig/config.go

```diff
@@ -193,11 +193,24 @@ func (config *Config) Validate(envName string) error {
 
 	// Check api models exist
 	modelNames := config.Models.Names()
+	allowNoEnv := false // if every API is path based, don't need env
 	for _, api := range config.APIs {
-		if !slices.HasString(modelNames, api.ModelName) {
+		if api.ModelPath == nil && api.ModelName == "" {
+			return errors.Wrap(ErrorSpecifyOnlyOneMissing("model_name", "model_path"), Identify(api))
```

Comment:
Oops, I just saw these, making my previous comment less relevant, which is why I usually send my comments all at once :) I still think it makes sense to put these validations in `apis.go`.